crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -852,14 +854,12 @@ where
                 .await?;
 
                 if !success {
-                    let error = format!(
-                        "Failed to recover the archiver for block number: {}",
-                        block_import_notification.block_number
-                    );
-                    error!(error);
-
                     return Err(sp_blockchain::Error::Consensus(sp_consensus::Error::Other(
```

Comment:
My expectation was that archiver would just try again. For that to happen we don't need to save block import notification, we can simply wait for next notification before doing re-initialization. And try that in a loop until it succeeds.

Reply:
This structure was primarily needed for override. Off top of my head I don't see the reason why this is needed and it does make code much harder to follow even if it works correctly.

Reply:
> If we don't save the problematic block notification then it would be lost and the archiving process would be broken.

It will not be if we wait for the next block import as I suggested. Archiver will restart in fully deterministic way and will continue operation just fine.

The issue with this line I commented on is that it returns an error, meaning archiver will exit and node will crash with an error. While I think archiver should just restart in a loop until it succeeds.